pallets/manta-pay/src/lib.rs

```diff
@@ -485,12 +502,24 @@ pub mod pallet {
         }
     }
 
-    impl<T> From<ReceiverPostError<ReceiverLedgerError>> for Error<T> {
+    impl<T> From<ReceiverPostError<ReceiverLedgerError<T>>> for Error<T>
+    where
+        T: Config,
+    {
         #[inline]
-        fn from(err: ReceiverPostError<ReceiverLedgerError>) -> Self {
+        fn from(err: ReceiverPostError<ReceiverLedgerError<T>>) -> Self {
             match err {
                 ReceiverPostError::AssetRegistered => Self::AssetRegistered,
-                ReceiverPostError::UnexpectedError(_) => Self::InternalLedgerError,
+                ReceiverPostError::UnexpectedError(e) => {
+                    match e {
+                        ReceiverLedgerError::ChecksumError
+                        | ReceiverLedgerError::MerkleTreeCapacityError => {
+                            T::Suspender::suspend_manta_pay_execution();
+                        }
+                        _ => {}
+                    };
+                    Self::InternalLedgerError
```

Comment:
Is it possible to do modify the definition of `InternalLedgerError` so we can return `Self::InternalLedgerError(e)` to keep track of the unexpected errors that don't shut down the ledger?

Reply:
Will take care of this in https://github.com/Manta-Network/Manta/issues/496